services/gateway/src/ops_batcher.rs

```diff
@@ -262,18 +283,156 @@ impl OpsBatcherRunner {
 
                     let tracker = self.tracker.clone();
                     let ids_for_receipt = ids.clone();
+                    let provider = self.provider.clone();
+                    let registry = self.registry;
                     tokio::spawn(async move {
                         match builder.get_receipt().await {
                             Ok(receipt) => {
                                 if receipt.status() {
-                                    tracker
-                                        .set_status_batch(
-                                            &ids_for_receipt,
-                                            RequestState::Finalized {
-                                                tx_hash: hash.clone(),
-                                            },
+                                    // Transaction succeeded. Since we set allowFailure=true in Multicall3,
+                                    // we need to inspect which subcalls (to the registry) failed.
+                                    // Try to debug_trace the transaction; if this fails (e.g. provider
+                                    // does not support debug), fall back to marking all as finalized.
```

Comment:
can we check if the provider supports debug trace before using `allowFailure: true`? 

We should avoid at all costs the possibility of setting something to Finalized that actually failed.